### PR TITLE
Stats: deprecate flag stats/new-stats-module-component

### PR DESCRIPTION
**Pull request by wilmersondasilva**
#### Proposed Changes

* Remove all usage of feature flag stats/new-stats-module-component.

#### Testing Instructions

* Spin up this change with the Calypso Live link.
* Navigate to the Stats page Traffic.
* Ensure everything on the page is the same as the previous one.

#### Pre-merge Checklist

- [ ] Has the general commit checklist been followed? (PCYsg-hS-p2)
- [ ] [Have you written new tests](https://wpcalypso.wordpress.com/devdocs/docs/testing/index.md) for your changes?
- [ ] Have you tested the feature in Simple (P9HQHe-k8-p2), Atomic (P9HQHe-jW-p2), and self-hosted Jetpack sites (PCYsg-g6b-p2)?
- [ ] Have you checked for TypeScript, React or other console errors?
- [ ] Have you used memoizing on expensive computations? More info in [Memoizing with create-selector](https://github.com/Automattic/wp-calypso/blob/trunk/packages/state-utils/src/create-selector/README.md) and [Using memoizing selectors](https://react-redux.js.org/api/hooks#using-memoizing-selectors) and [Our Approach to Data](https://github.com/Automattic/wp-calypso/blob/trunk/docs/our-approach-to-data.md)
- [ ] Have we added the "[Status] String Freeze" label as soon as any new strings were ready for translation (p4TIVU-5Jq-p2)?
- [ ] For changes affecting Jetpack: Have we added the "[Status] Needs Privacy Updates" label if this pull request changes what data or activity we track or use (p4TIVU-ajp-p2)?

Related to #70210

### CODE REVIEW

File: client/my-sites/stats/site.jsx [hunk 1]
```diff
@@ -330,7 +328,6 @@ class StatsSite extends Component {
 							query={ query }
 							statType="statsTopPosts"
 							showSummaryLink
-							showNewModules={ showNewModules }
```
Comment (grzegorz-cp) [inline client/my-sites/stats/site.jsx:296]: I'm afraid that for now, we can't remove any of the `showNewModules` component parameters. However, we can remove the feature flag and keep the parameter as `true`. For example:
```
                       <StatsModule
							path="posts"
							moduleStrings={ moduleStrings.posts }
							period={ this.props.period }
							query={ query }
							statType="statsTopPosts"
							showSummaryLink
							showNewModules
					/>
```

The component spills to the `Insights` page, and `showNewModules` was a way to limit it to the `Traffic` page. 
Once we add horizontal bars to both pages, we can remove this property. There is probably a problem parsing some data on the `Insights` page.

File: client/my-sites/stats/stats-module/index.jsx [hunk 2]
```diff
@@ -156,8 +154,7 @@ class StatsModule extends Component {
 			'is-refreshing': requesting && ! isLoading,
 		} );
 
-		const shouldShowNewModule =
```
Comment (grzegorz-cp) [inline client/my-sites/stats/stats-module/index.jsx:159]: It's the same here, we can remove the feature flag but keep:
```
const shouldShowNewModule = showNewModules && ! summary;
```

**Review by grzegorz-cp — CHANGES_REQUESTED** [review]
Thanks for looking into this. We still need to control the use of horizontal bar components and limit it from showing up on the `Insights` page. While the feature flag can be removed because the feature was shipped on the `Traffic` page, we need to keep the `showNewModules` parameter (always `true` on the `Traffic` page and not controlled by a flag). Otherwise, the `Insights` page is broken.

File: client/my-sites/stats/site.jsx (hunk 1 above)
Comment (jsnmoon) [reply client/my-sites/stats/site.jsx:296]: Grzegorz is right; this approach reintroduces the regression we squashed in #69942.

**Comment by wilmersondasilva** [comment]
@grzegorz-cp thanks for reviewing it and for pointing out this. I didn't know. I just implemented the changes requested and it should be fine now.

**Review by jsnmoon — APPROVED** [review]
Changes look reasonable and work as expected on the live branch 👍 

@grzegorz-cp: Is there any follow-up code removal we need to do for this feature flag?

**Comment by grzegorz-cp** [comment]
> @grzegorz-cp: Is there any follow-up code removal we need to do for this feature flag?

I'll add a task to our maintenance board to remove the additional property once all horizontal bar modules are in place.

**Review by grzegorz-cp — APPROVED** [review]
Looks good :)

**Pull request merged**